lib/serverengine/daemon.rb

```diff
@@ -47,14 +47,23 @@ def initialize(server_module, worker_module, load_config_proc={}, &block)
       @chuser = @config[:chuser]
       @chgroup = @config[:chgroup]
       @chumask = @config[:chumask]
+
+      @pid = nil
+      @command_pipe = @config.fetch(:command_pipe, nil)
+      @command_sender = @config.fetch(:command_sender, "signal")
```

Comment:
> Signals don't work on Windows at all, is that right?

Yes.
To sending signals to other processes, only :INT (often ignored by target processes by various reasons) and :KILL are supported on ruby on Windows.

> So it sounds better that here uses "pipe" as the default value on Windows.

At first, I thought that we had to keep compatibility with already existing ServerEngine scripts.
But, rethinking now, there are no scripts which correctly be run on Windows :P
So I agree.